tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoServiceFactory.cs

```diff
@@ -0,0 +1,126 @@
+    /// <param name="gitHelper">Git helper instance for repository operations</param>
+    /// <param name="logger">Optional logger instance for diagnostics</param>
+    /// <returns>Language-specific repository service</returns>
+    public static ILanguageRepoService CreateService(string packagePath, IGitHelper gitHelper, ILogger? logger = null)
+    {
+        logger?.LogInformation($"Create service for package at: {packagePath}");
+        if (string.IsNullOrWhiteSpace(packagePath))
+        {
+            throw new ArgumentException("Package path cannot be null or empty", nameof(packagePath));
+        }
+
+        if (!Directory.Exists(packagePath))
+        {
+            throw new DirectoryNotFoundException($"Package path does not exist: {packagePath}");
+        }
+
+        // Discover the repository root from the project path
+        var repoRootPath = gitHelper.DiscoverRepoRoot(packagePath);
+        logger?.LogInformation($"Discovered repository root: {repoRootPath}");
+        
+        // Create language service using factory (detects language automatically)
+        logger?.LogInformation($"Creating language service for repository at: {repoRootPath}");
+        var detectedLanguage = DetectLanguage(repoRootPath, logger);
+
+        return detectedLanguage switch
+        {
+            "python" => new PythonLanguageRepoService(packagePath, gitHelper, logger),
+            "javascript" => new JavaScriptLanguageRepoService(packagePath),
+            "dotnet" => new DotNetLanguageRepoService(packagePath),
+            "go" => new GoLanguageRepoService(packagePath),
+            "java" => new JavaLanguageRepoService(packagePath),
+            _ => new LanguageRepoService(packagePath) // Base implementation for unsupported languages
+        };
+    }
+
+    /// <summary>
+    /// Detects the primary language of a repository based on file patterns and configuration files.
+    /// First checks for Language-Settings.ps1 as mentioned in the gist, then falls back to file-based detection.
+    /// </summary>
+    /// <param name="repositoryPath">Path to the repository root</param>
+    /// <param name="logger">Optional logger instance for diagnostics</param>
+    /// <returns>Detected language string</returns>
+    public static string DetectLanguage(string repositoryPath, ILogger? logger = null)
+    {        
+        // First, try to detect from eng/scripts/Language-Settings.ps1 as mentioned in the gist
+        var languageSettingsPath = Path.Combine(repositoryPath, "eng", "scripts", "Language-Settings.ps1");
+        logger?.LogInformation($"Language settings path {languageSettingsPath}");
+        if (File.Exists(languageSettingsPath))
+        {
+            logger?.LogDebug("Found Language-Settings.ps1 file at: {LanguageSettingsPath}", languageSettingsPath);
+            try
+            {
+                var content = File.ReadAllText(languageSettingsPath);
```

Comment:
Can we not rely on the directory or git remote name, or maybe even the readme header? Scanning a whole powershell script seems like it's maybe not the best source of truth.

Reply:
I'll use the readme